SingularityBase/src/main/java/com/hubspot/singularity/RequestUtilization.java

```diff
@@ -0,0 +1,62 @@
+package com.hubspot.singularity;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class RequestUtilization {
+  private final String requestId;
+  private final String deployId;
+  private long memBytesTotal = 0;
+  private double cpuTotal = 0;
+  private int numTasks = 0;
+
+  @JsonCreator
+  public RequestUtilization(@JsonProperty("requestId") String requestId,
+                            @JsonProperty("deployId") String deployId) {
+    this.requestId = requestId;
+    this.deployId = deployId;
+  }
+
+  public RequestUtilization addMemBytes(long memBytes) {
+    this.memBytesTotal += memBytes;
+    return this;
+  }
+
+  public RequestUtilization addCpu(double cpu) {
+    this.cpuTotal += cpu;
+    return this;
+  }
+
+  public RequestUtilization incrementTaskCount() {
+    this.numTasks++;
+    return this;
+  }
+
+  public long getMemBytesTotal() {
+    return memBytesTotal;
+  }
+
+  public double getCpuTotal() {
+    return cpuTotal;
+  }
+
+  public int getNumTasks() {
+    return numTasks;
+  }
+
+  public double getAvgMemBytesUsed() {
```

Comment:
having these getters here means they would be included as json fields. Since they are easily calculated, and not used anywhere else in the code maybe exclude these? (think two extra fields x 3000+ requests all in one json blob)